src/modules/blogs/blog.controller.ts

```diff
@@ -0,0 +1,121 @@
+import { NextFunction, Request, Response } from "express";
+import { BlogService } from "./blog.service";
+import { AuthRequest } from "../../types";
+
+export class BlogController {
+  static async createBlogHandler(
+    req: AuthRequest,
+    res: Response,
+    next: NextFunction,
+  ) {
+    try {
+      const user = req.user;
+      const blog = await BlogService.create(user.userId, req.body);
+      res.status(201).json({
+        message: "Blog created successfully",
+        data: blog,
+      });
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  static async getMyBlogsHandler(
+    req: AuthRequest,
+    res: Response,
+    next: NextFunction,
+  ) {
+    try {
+      const take = req.query?.take || 20;
+      const skip = req.query?.skip || 0;
+      const user = req.user;
+      const blogs = await BlogService.getMyBlogs(
+        user.userId,
+        take as number,
+        skip as number,
+      );
+      res.status(200).json({
+        message: "Blogs fetched successfully",
+        data: blogs,
+      });
+    } catch (err) {
+      next(err);
+    }
+  }
+  static async getPublicBlogsHandler(
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ) {
+    try {
+      const take = req.query?.take || 20;
+      const skip = req.query?.skip || 0;
+      const blogs = await BlogService.getAllBlogs(
+        take as number,
+        skip as number,
+      );
+      res.status(200).json({
+        message: "Blogs fetched successfully",
+        data: blogs,
+      });
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  static async getBlogByIdHandler(
+    req: Request,
+    res: Response,
+    next: NextFunction,
```

Comment:
why do you need next function here

Reply:
To pass AppError to next middleware ( error handler )